activerecord/lib/active_record/migration.rb

```diff
@@ -1067,7 +1067,7 @@ def parse_migration_filename(filename) # :nodoc:
       def migrations(paths)
         paths = Array(paths)
 
-        files = Dir[*paths.map { |p| "#{p}/**/[0-9]*_*.rb" }]
+        files = Dir[*paths.map { |p| migration_paths_regex(p) }]
```

Comment:
That would work as well. My initial thinking was to make as little change as possible but it seems that's going to be impossible. I'm going to wait until I have the time to unravel the db:migrate:status issue also before I take a side. If we can reuse this method there as well, which I suspect we can, then I'm all for changing it.